pym/bob/input.py

```diff
@@ -1852,6 +1867,12 @@ def getProvideDepsResolver(pattern):
     else:
         return VerbatimProvideDepsResolver(pattern)
 
+def getLayerName(layerSpec):
+    if isinstance(layerSpec, str):
+        return layerSpec
+    else:
+        return layerSpec.get('name')
+
```

Comment:
That should be unnecessary once the `LayerValidator` always creates `LayerSpec` objects.